File: Domain/ProductPrice.cs
using System.ComponentModel.DataAnnotations;
using Newtonsoft.Json;
using Newtonsoft.Json.Serialization;

namespace Domain;
[JsonObject(NamingStrategyType = typeof(SnakeCaseNamingStrategy))]

public class ProductPrice
{
    public string ProductPriceId { get; set; } = null!;
    public string ProductId { get; set; } = null!;
    public string ProductPriceTypeId { get; set; } = null!;
    public string? ProductPricePurposeId { get; set; } = null!;
    public string CurrencyUomId { get; set; } = null!;
    public string? ProductStoreGroupId { get; set; } = null!;
    public DateTime FromDate { get; set; }
    public DateTime? ThruDate { get; set; }
    public decimal? Price { get; set; }
    public string? TermUomId { get; set; }
    public string? CustomPriceCalcService { get; set; }
    public decimal? PriceWithoutTax { get; set; }
    public decimal? PriceWithTax { get; set; }
    public decimal? TaxAmount { get; set; }
    public decimal? TaxPercentage { get; set; }
    public string? TaxAuthPartyId { get; set; }
    public string? TaxAuthGeoId { get; set; }
    public string? TaxInPrice { get; set; }
    public DateTime? CreatedDate { get; set; }
    public string? CreatedByUserLogin { get; set; }
    public DateTime? LastModifiedDate { get; set; }
    public string? LastModifiedByUserLogin { get; set; }
    public DateTime? LastUpdatedStamp { get; set; }
    public DateTime? LastUpdatedTxStamp { get; set; }
    public DateTime? CreatedStamp { get; set; }
    public DateTime? CreatedTxStamp { get; set; }


    public UserLogin? CreatedByUserLoginNavigation { get; set; }
    public Uom? CurrencyUom { get; set; } = null!;
    public CustomMethod? CustomPriceCalcServiceNavigation { get; set; }
    public UserLogin? LastModifiedByUserLoginNavigation { get; set; }
    public Product? Product { get; set; } = null!;
    public ProductPricePurpose? ProductPricePurpose { get; set; } = null!;
    public ProductPriceType? ProductPriceType { get; set; } = null!;
    public ProductStoreGroup? ProductStoreGroup { get; set; } = null!;
    public Geo? TaxAuthGeo { get; set; }
    public Party? TaxAuthParty { get; set; }
    public Uom? TermUom { get; set; }
}

File: client-app/src/app/models/product/productPrice.ts
export interface ProductPrice {
    id: string;
    productId: string;
    productPriceTypeId: string;
    productPricePurposeId: string | null;
    currencyUomId: string;
    productStoreGroupId: string | null;
    fromDate: Date;
    thruDate: Date | null;
    price: number | null;
    productPriceTypeDescription: string | null;
    currencyUomDescription: string | null;
    termUomId: string | null;
    customPriceCalcService: string | null;
    priceWithoutTax: number | null;
    priceWithTax: number | null;
    taxAmount: number | null;
    taxPercentage: number | null;
    taxAuthPartyId: string | null;
    taxAuthGeoId: string | null;
    taxInPrice: string | null;
    createdByUserLogin: string | null;
    lastModifiedByUserLogin: string | null;
    lastUpdatedStamp: Date | null;
    createdStamp: Date | null;
    rowVersion: any;
}
